vector/src/main/res/values/strings.xml

```diff
@@ -425,6 +425,8 @@
     <string name="permissions_msg_contacts_warning_other_androids">Element can check your address book to find other Matrix users based on their email and phone numbers.\n\nDo you agree to share your address book for this purpose?</string>
 
     <string name="permissions_action_not_performed_missing_permissions">Sorry. Action not performed, due to missing permissions</string>
+    <string name="permissions_denied_qr_code">To scan a QR code, allow Camera permission to take a picture</string>
+    <string name="permissions_denied_add_contact">To check your address book, allow Contact permission.</string>
```

Comment:
And
"Please grant the Contact permission to check your address book"

Reply:
Some ideas:

- "To check your address book, allow access to your contacts"  
- "To check your contacts, allow access to your contacts"
- "To check your contacts, you need to allow access to your contacts"
- "To check your contacts, you need to allow access to your address book"

I think saying contacts first is good because the label they tapped on says "contacts book"?

Reply:
Yes. I just want to tell that those 2 messages are displayed only if the user had denied the permission request in the past.